firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionTesterApiClient.java

```diff
@@ -109,62 +96,116 @@ Task<AppDistributionReleaseInternal> fetchNewRelease() {
               String.format(
                   "v1alpha/devices/-/testerApps/%s/installations/%s/releases",
                   firebaseApp.getOptions().getApplicationId(), fid);
-          String responseBody = makeGetRequest(path, token);
-          return parseNewRelease(responseBody);
+          HttpResponse response = testerApiHttpClient.makeGetRequest(path, token);
+          if (response.code() == 404) {
```

Comment:
Why is this special and not handled by `getExceptionForHttpResponse()`?
Also, I think 404 should be exclusively returned for "not found" - is this a bug on the server-side? If so, we ought to fix that.

Reply:
Cleaned this up according to our discussion. All error handling is done in the TesterApiHttpClient, with custom tag prepended to error messages.

Reply:
RE: NOT_FOUND, currently we do return 404 for missing access cases, like when [the tester doesn't have access to the app](http://google3/java/com/google/firebase/appdistro/boq/testers/v1alpha/service/actions/ListTesterAppInstallationReleasesAction.java;l=278;rcl=452123374). Also if the app doesn't exist we also return 404, but we should return 403 according to https://google.aip.dev/211.

We should fix this on the backend (assuming it won't break any old SDKs, which it might), but in the meantime I think we need to keep treating 404s as `AUTHENTICATION_FAILURE`.